### PR TITLE
Add Wi-Fi onboarding using Bluetooth

### PR DESCRIPTION
**Pull request by grunskis**
Adds Wi-Fi on-boarding through BLE (implemented by Tim) with some small adaptations.

Worked for me on iOS, but not on Android. On Android the hub discovery works, but subscription to any characteristic fails. I'm running Android 5.1, which might be the issue...

### CODE REVIEW

File: senic_hub/setup_app/HubOnboarding.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+      return "down"
+    } else if (this.connectionState === WifiConnectionState.CONNECTION_STATE_DISCONNECTED) {
+      return "disconnected"
+    } else if (this.connectionState === WifiConnectionState.CONNECTION_STATE_CONNECTING) {
+      return "connecting"
+    } else if (this.connectionState === WifiConnectionState.CONNECTION_STATE_CONNECTED) {
+      return "connected"
+    } else {
+      return "unknown"
+    }
```
Comment (larsblumberg) [inline senic_hub/setup_app/HubOnboarding.js]:
```
return ({
  WifiConnectionState.CONNECTION_STATE_DOWN: 'down',
  WifiConnectionState.CONNECTION_STATE_DISCONNECTED: 'disconnected',
  WifiConnectionState.CONNECTION_STATE_CONNECTING: 'connecting',
  WifiConnectionState.CONNECTION_STATE_CONNECTED: 'connected',
})[this.connectionState] || "unknown"
```

File: senic_hub/setup_app/HubOnboarding.js [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    if (characteristic.value.length < 4) {
+      console.log("Hub: received empty connection state.")
+      return;
+    }
+    let rawState = base64.decode(characteristic.value)
+    this.connectionState = rawState.charCodeAt(0)
+    if (this.connectionState > 1 && rawState.length > 1) {
+      this.currentSsid = rawState.slice(1)
+      console.log("Hub: received connection state: " + this.getConnectionStateString() + " to " + this.currentSsid)
+    } else {
```
Comment (larsblumberg) [inline senic_hub/setup_app/HubOnboarding.js]: Let's put `else` into the next line:

    }
    else {

File: senic_hub/setup_app/android/app/src/main/java/com/senichubsetup/MainApplication.java [hunk 3]
```diff
@@ -17,7 +18,8 @@ public boolean isDebug() {
     protected List<ReactPackage> getPackages() {
         return Arrays.<ReactPackage>asList(
             new RNNetworkInfoPackage(),
-            new VectorIconsPackage()
+            new VectorIconsPackage(),
+            new BlePackage()
```
Comment (larsblumberg) [inline senic_hub/setup_app/android/app/src/main/java/com/senichubsetup/MainApplication.java]: Can we order alphabetically?

File: senic_hub/setup_app/components/SetupHub.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  startScanning() {
+    this.manager.startDeviceScan(null, null, (error, device) => {
+      if (error) {
+        alert("An error occurred while scanning: " + JSON.stringify(error))
+        console.error("An error occurred while scanning:", error)
+        return
+      }
+      //console.log("Found BLE device:", device)
```
Comment (larsblumberg) [inline senic_hub/setup_app/components/SetupHub.js]: Delete comment?

File: senic_hub/setup_app/components/SetupWifi.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+          id: 'skip',
+          disabled: false,
+          onPress: () => this.pushScreen('setup.nuimo')
+        }])
+      }
+    })
+  }
+
+  willDisappear() {
+
```
Comment (larsblumberg) [inline senic_hub/setup_app/components/SetupWifi.js]: Put TODO comment here to do some clean up? Or delete method?

File: senic_hub/setup_app/components/SetupWifiPassword.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+          onChangeText={(password) => this.setState({password})}
+          value={this.state.password}
+          placeholder="Password"
+          secureTextEntry={true}
+        />
+
+        <Button
+          onPress={() => { this.onSubmit() }}
+          title="Submit"
+           accessibilityLabel="Submit Password"
```
Comment (larsblumberg) [inline senic_hub/setup_app/components/SetupWifiPassword.js]: One leading space too much. I guess the linter is calling us soon to offer some service to us :)

File: senic_hub/setup_app/components/SetupWifiPassword.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        flex: 1,
+        flexDirection: 'column',
+        justifyContent: 'space-between',
+    },
+    title: {
+        fontSize: 18,
+        textAlign: 'center',
+        margin: 10,
+    },
+})
```
Comment (larsblumberg) [inline senic_hub/setup_app/components/SetupWifiPassword.js]: Let's use only 2 spaces for indentation

File: senic_hub/setup_app/HubOnboarding.js (hunk 1 above)
Comment (grunskis) [inline senic_hub/setup_app/HubOnboarding.js]: This apparently is not valid Javascript, dict keys are not allowed to have dot in the name

**Comment by grunskis** [comment]
@larsblumberg Addressed all the points

**Pull request merged**